test/integration/options/watch.spec.js

```diff
@@ -47,6 +47,23 @@ describe('--watch', function() {
       });
     });
 
+    it('reruns test when watched test file is crashed', function() {
+      const testFile = path.join(tempDir, 'test.js');
+      copyFixture(DEFAULT_FIXTURE, testFile);
+
+      return runMochaWatchJSONAsync([testFile], tempDir, () => {
+        replaceFileContents(testFile, 'done();', 'done((;');
+      })
+        .then(() => {
+          return runMochaWatchJSONAsync([testFile], tempDir, () => {
+            replaceFileContents(testFile, 'done((;', 'done();');
+          });
+        })
+        .then(results => {
+          expect(results, 'to have length', 1);
+        });
```

Comment:
It's not necessary to call `runMochaWatchJSONAsync`  twice. You don't process the result of the first call (good, then bad test) anyway. `results` contains only the second call (bad, then good test).

The following runs first the bad, then the good test with only one call:
```js
it('reruns test when watched test file crashes', function() {
      const testFile = path.join(tempDir, 'test.js');
      copyFixture(DEFAULT_FIXTURE, testFile);

      replaceFileContents(testFile, 'done();', 'done((;');

      return runMochaWatchJSONAsync([testFile], tempDir, () => {
        replaceFileContents(testFile, 'done((;', 'done();');
      })
        .then(results => {
          expect(results, 'to have length', 1);
        });
    });
```

Reply:
Smart approach!!